### DISCUSSION TITLE
Why doesn't wait_window() work?

### DISCUSSION THREAD

**Discussion by mxl0525** [israel-dryer/ttkbootstrap#240] [Q&A]
@israel-dryer 
import ttkbootstrap as ttk

if __name__ == '__main__':

    app = ttk.Window()
    app.title('app')
    app.wait_window()

    root = ttk.Window()
    root.title('test')
    root.mainloop()

**Comment by israel-dryer** [comment]
That's strange. You're right, it should work as written, though I've never seen it used in this way. What are you trying to achieve?

Typically this method is implemented as in the example below in order to create a local event loop that waits for a window or widget to be destroyed before continuing.

```python
import ttkbootstrap as ttk 

app = ttk.Window()

def show_my_popup():
    top = ttk.Toplevel(title="top")
    top.wait_window()
    print("Hello world")

ttk.Button(app, text="Show popup", command=show_my_popup).pack()

app.mainloop()
```

In your example, you are creating a local event loop before the main event loop has even been called. So, I'm not sure why it should work conceptually, even though it does with regular tk.

I'll look into it, but from what I can tell, this method still works when implemented in a way such as what I've used above when the `wait_window` method is called after the main event loop is running.

**Reply by mxl0525** [reply]
Thanks, but tkinter is ok.



import tkinter as tk

if __name__ == '__main__':

    app = tk.Tk()
    app.title('app')
    app.wait_window()

    root = tk.Tk()
    root.title('test')
    root.mainloop()

**Comment by rdbende** [comment]
Strange. For me it works just fine if I use the `tkinter.Tk` instead of `ttkboostrap.Window`. But anyways I don't think you should use two `ttkbootstrap.Window` in one program.

**Comment by TheDemonGuard** [comment]
Seems the issue is resolved with the latest ttkbootstrap version (1.14.4)
